update font, background colors, replace hardcoded values

evwilkin opened the pull request:
Closes #3598 

This PR addresses bugs with dark theme text & background colors across Colors for charts, Spacers, and Colors pages.

patternfly-build commented:
Preview: https://patternfly-org-pr-3607-site.surge.sh

evwilkin commented:
@mcoker - tagged you for review because I'm updating the CSS variable names displayed on the Colors for charts page to the new `pf-v5-` values, and also using those values for each color circle example.  Want to make sure that looks correct, and if anything around the charts colors needs to be adjusted for dark mode.

evwilkin commented:
@mcarrano thanks for those notes
- I've fixed the text color on the swatches on the Colors page
- the "Considering line height and padding" screenshot above - the dark text is actually part of an image along with the the white/green padding lines beside it.  I've added a white background behind the image to maintain the color contrast of the text in the image.

mcoker commented on packages/documentation-site/patternfly-docs/content/design-guidelines/styles/colors-for-charts/colors-grid.js:
```diff
@@ -12,7 +12,7 @@ const ColorsGrid = () => (
         <GridItem span={10}>
           <div className="ws-content-chart-colors-gallery--information">
             <div>#8BC1F7</div>
```
This is going to change between light/dark theme. I'm not sure what a good fix is. I don't think we have tokens currently for dark theme values - either a dark theme value in the current/existing token, or potentially a separate token for dark theme - but if we did, this whole grid bit could probably just loop through the chart color tokens and spit out the var/hex color from the token, and include the dark theme hex color and hide it in light theme and show it in dark theme?

mcoker reviewed (approved):
LGTM. Left a comment about the dark theme color values not updating. Here are the dark theme color changes we should see https://github.com/patternfly/patternfly/blob/main/src/patternfly/base/themes/dark/_chart-globals.scss

That said, the colors aren't updating in dark theme as they should. Looks like a var name issue in core - https://github.com/patternfly/patternfly/pull/5673

mcarrano reviewed (approved):
Looks good. Thanks for the updates @evwilkin !

evwilkin commented:
@mcoker thanks for the feedback - I've updated this to import the Charts stylesheet, and to grab the CSS variable values directly from the root element and loop through them as you suggested.  I've also added a mutation observer that listens for the class to change on the root element & get the updated CSS var values, to update when the dark theme class is toggled.

mcoker commented on packages/documentation-site/patternfly-docs/content/design-guidelines/styles/spacers/spacers.css:
```diff
@@ -57,23 +57,23 @@
 
 .ws-content-correct {
-  color: #486b00;
+  color: var(--pf-v5-global--info-color--100);
```
Did you mean to turn this content blue? It was green before. 

```suggestion
  color: var(--pf-v5-global--success-color--100);
```

mcoker commented on packages/documentation-site/patternfly-docs/content/design-guidelines/styles/colors-for-charts/colors-grid.js:
```diff
@@ -1,409 +1,83 @@
 import React from 'react';
 import { Grid, GridItem } from '@patternfly/react-core';
+import '@patternfly/patternfly/patternfly-charts-theme-dark.css';
```
Just a nit, but should we move the base charts CSS import into this file, too, just for organization? Or does it need to be in the other file? https://github.com/patternfly/patternfly-org/blob/d93dc163789d545df723003a5e5604b4582971e2/packages/documentation-site/patternfly-docs/content/design-guidelines/styles/colors-for-charts/colors-for-charts.md?plain=1#L14

```suggestion
import '@patternfly/patternfly/patternfly-charts.css';
import '@patternfly/patternfly/patternfly-charts-theme-dark.css';
```

mcoker commented on packages/documentation-site/patternfly-docs/content/design-guidelines/styles/colors-for-charts/colors-grid.js:
```diff
@@ -1,409 +1,83 @@
+  {headingText: 'Gold', varColor: 'gold'},
+  {headingText: 'Orange', varColor: 'orange'},
+  {headingText: 'Red', varColor: 'red'},
+  {headingText: 'Grey', varColor: 'black'}
```
I noticed in the main design guidelines color page, we're using "gray", and using "grey" here. Though I pinged @mceledonia about using "black" here instead, since that's the color name in the code, and also what designers tend to give developers (for example, "black-300" and "green-300"), and he suggested updating the wording to reflect the actual color name - so maybe we can change this to "black"? cc @mceledonia @mmenestr can you confirm?

mcoker reviewed (dismissed):
LGTM! Left a few comments, the only one I'm really interested in was the change of the green to blue text for the "correct" block.

Also if we ever wanted to update the layout a bit to space the color/var/value pair a bit, I mocked that up with just the blue colors in a way that might work well. Everything is just using the flex/grid layout classes, except for removing the margin from `.ws-content-chart-colors-gallery--information` as that shouldn't be necessary.

<img width="1177" alt="Screenshot 2023-06-23 at 1 40 29 PM" src="https://github.com/patternfly/patternfly-org/assets/35148959/69ac00e3-e9da-45d1-80db-3873149e9d1d">

mcarrano reviewed (approved):
LGTM. Thanks @evwilkin !

evwilkin commented:
@mcoker pulled in your suggestions, thanks!
